packages/ember-simple-auth/addon/session-stores/adaptive.js

```diff
@@ -112,28 +132,25 @@ export default Base.extend({
       this._fastboot = owner.lookup('service:fastboot');
     }
 
-    this._isLocalStorageAvailable = this.hasOwnProperty('_isLocalStorageAvailable') ? this._isLocalStorageAvailable : testLocalStorageAvailable();
-
     let store;
     if (this.get('_isLocalStorageAvailable')) {
+      store = this.get('localStorage');
       const options = { key: this.get('localStorageKey') };
       options._isFastBoot = false;
-      store = this._createStore(LocalStorage, options);
+      store.setProperties(options);
     } else {
-      const options = this.getProperties('cookieDomain', 'cookieName', 'cookieExpirationTime', 'cookiePath');
-      options._fastboot = this.get('_fastboot');
-      options._cookies = this.get('_cookies');
+      store = this.get('cookie');
+      const options = this.getProperties('sameSite', 'cookieDomain', 'cookieName', 'cookieExpirationTime', 'cookiePath');
 
-      store = this._createStore(Cookie, options);
+      store._initialize(options);
```

Comment:
Does

```suggestion
      store.setProperties(options);
```

not work here?

Reply:
I could check that out, although `_initalize` currently needs to treat the `cookieName` property in a special way.

The proxy properties that were receiving some default values on initialization weren't causing any issues before, but right now since we can't really pass any properties during init, but we set them after instead - its causing e.g `cookieName` to be cached with a wrong value.